core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -516,20 +501,23 @@ class KafkaApis(val requestChannel: RequestChannel,
       val response = new FetchResponse(fetchedPartitionData, 0)
       val responseStruct = response.toStruct(versionId)
 
-      def fetchResponseCallback(throttleTimeMs: Int) {
-        trace(s"Sending fetch response to client $clientId of ${responseStruct.sizeOf} bytes.")
-        val responseSend = response.toSend(responseStruct, throttleTimeMs, request.connectionId, request.header)
-        requestChannel.sendResponse(new RequestChannel.Response(request, responseSend))
+      def fetchResponseCallback(bandwidthThrottleTimeMs: Int) {
+        def sendResponseCallback(requestThrottleTimeMs: Int) {
+          trace(s"Sending fetch response to client $clientId of ${responseStruct.sizeOf} bytes.")
+          val responseSend = response.toSend(responseStruct, bandwidthThrottleTimeMs + requestThrottleTimeMs, request.connectionId, request.header)
+          requestChannel.sendResponse(new RequestChannel.Response(request, responseSend))
+        }
+        sendResponseMaybeThrottle(request, request.header.clientId, sendResponseCallback)
```

Comment:
Hmm, this can be a bit tricky. Fetch requests from the follower are considered internal and shouldn't be throttled.